### ISSUE TITLE
Tool is hanging due to loose grep expression

### ISSUE THREAD

**Issue by IzhakJakov** [izhakjakov/getGoDependencyHierarchy#2]
### Steps to reproduce
```
 $ git clone 'https://github.com/helm/helm.git'`

 $ cd helm

 $ git describe --tags
v3.8.0-rc.1-11-g59dbda91`

 $ getGoDependencyHierarchy 'github.com/dgrijalva/jwt-go@v3.2.0+incompatible'
<hanging>
```

**Comment by IzhakJakov** [comment]
The tool is looking for a dependent of `github.com/containerd/containerd@v1.5.0-rc.0` and finds `github.com/containerd/imgcrypt@v1.1.1` which is correct.
Then, it is looking for a dependent of `github.com/containerd/imgcrypt@v1.1.1` and finds `github.com/containerd/containerd@v1.5.0-rc.0` which is wrong.
  
The issue is that `github.com/containerd/containerd@v1.5.0-rc.0` is a dependent `github.com/containerd/imgcrypt@v1.1.1-0.20210312161619-7ed62a527887` and the tool using grep which is looking for a "word" that matches `github.com/containerd/imgcrypt@v1.1.1`
`github.com/containerd/imgcrypt@v1.1.1-0.20210312161619-7ed62a527887` is considered to be 3 words separated by dash (`-`).
  
This is why the tool thinks that `github.com/containerd/containerd@v1.5.0-rc.0` is a dependent of `github.com/containerd/imgcrypt@v1.1.1`

**Comment by IzhakJakov** [comment]
Fixed in commit 09919594bb0732a1ed6b8d2034cb0a3bdca1f348.
